File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/tutorial/Tutorial.java
/*
 * SPDX-FileCopyrightText: 2020 DB Station&Service AG <bahnhoflive-opensource@deutschebahn.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package de.deutschebahn.bahnhoflive.tutorial;

public class Tutorial {

    public String id; // view
    public String title;
    public String descriptionText;

    public int currentCount = Integer.MAX_VALUE;
    public int countdown = Integer.MAX_VALUE; // until show...

    public boolean closedByUser = false;

    public String getId() {
        return id;
    }

    public Tutorial(String id, String title, String descriptionText, int countdown) {
        this.id = id;
        this.title = title;
        this.descriptionText = descriptionText;
        this.countdown = countdown;
        this.currentCount = countdown;
    }

    @Override
    public String toString() {
        return "Tutorial{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                ", descriptionText='" + descriptionText + '\'' +
                ", currentCount=" + currentCount +
                ", countdown=" + countdown +
                ", closedByUser=" + closedByUser +
                '}';
    }
}

File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/tutorial/Tutorial.kt
/*
 * SPDX-FileCopyrightText: 2020 DB Station&Service AG <bahnhoflive-opensource@deutschebahn.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
package de.deutschebahn.bahnhoflive.tutorial

class Tutorial(// view
    @JvmField var id: String, var title: String, @JvmField var descriptionText: String, countdown: Int
) {
    @JvmField
    var currentCount = Int.MAX_VALUE
    @JvmField
    var countdown = Int.MAX_VALUE // until show...
    @JvmField
    var closedByUser = false

    init {
        this.countdown = countdown
        currentCount = countdown
    }

    override fun toString(): String {
        return "Tutorial{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                ", descriptionText='" + descriptionText + '\'' +
                ", currentCount=" + currentCount +
                ", countdown=" + countdown +
                ", closedByUser=" + closedByUser +
                '}'
    }
}
